lib/bolt/config/transport/base.rb

```diff
@@ -11,39 +11,42 @@ class Base
 
         def initialize(data = {}, boltdir = nil)
           assert_hash_or_config(data)
-          @input   = reference?(data) ? data : filter(data)
-          @config  = Bolt::Util.deep_merge(defaults, @input)
-          @boltdir = boltdir
-          validate
+          @input    = data
+          @resolved = !Bolt::Util.references?(input)
+          @config   = Bolt::Util.deep_merge(defaults, filter(input))
```

Comment:
Should we even do this merge if the input is unresolved? That would help ensure that it's impossible to _use_ unresolved data.

Reply:
Definitely went back and forth on whether or not to do that. Updated to default to `defaults` if the input is unresolved.